src/main/java/com/google/sps/servlets/LatestRating.java

```diff
@@ -0,0 +1,53 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.json.JSONObject;
+
+@WebServlet("/latest-rating")
+public class LatestRating extends HttpServlet {
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    JSONObject previousRatingInfo = new JSONObject();
+    Key parentTermKey = KeyFactory.stringToKey((String) request.getParameter("term-key"));
+    String userId = (String) request.getParameter("reviewer-id");
+    List<Entity> latestRatingQuery = getLatestRating(userId, parentTermKey);
+
+    if (!latestRatingQuery.isEmpty()) {
+      Entity previousRatingEntity = latestRatingQuery.get(0);
+      previousRatingInfo.put("termInput", previousRatingEntity.getProperty("comments-term"));
+      previousRatingInfo.put("profInput", previousRatingEntity.getProperty("comments-professor"));
+      previousRatingInfo.put("difficulty", previousRatingEntity.getProperty("difficulty"));
+      previousRatingInfo.put("hours", previousRatingEntity.getProperty("hours"));
+      previousRatingInfo.put("ratingTerm", previousRatingEntity.getProperty("perception-term"));
+      previousRatingInfo.put(
+          "ratingProf", previousRatingEntity.getProperty("perception-professor"));
+      previousRatingInfo.put("grade", previousRatingEntity.getProperty("grade"));
+      previousRatingInfo.put("translation", previousRatingEntity.getProperty("translation"));
```

Comment:
is there a reason you don't name the properties the same thing and set the entities equal to each other? or is there a reason that wouldn't work

Reply:
So in this case, I was trying to make the names similar to how they are inputted in the actual html file (which is with term input, prof input, etc). So I was trying to map from this one to the JS one.

Reply:
But to be honest, it would be easier if they could all be changed to match everywhere so will check with Dagmawi in case he uses this Entity somewhere.

Reply:
Talked with Dagmawi and we decided that for now we would stick with how they are formatted for now to avoid confusion